### PR TITLE
120 implement SaveableEntity for inventory + fix warnings and bugs

### PR DESCRIPTION
**Pull request by juhwansonn**

### CODE REVIEW

File: src/main/java/save/SaveInventory.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+public class SaveInventory implements SavableEntity {
+    final Inventory inventory = Player.getInventory();
+
+    /**
+     * @return a string representation of inventory to be saved
+     */
+    @Override
+    public String toSavableString() {
+        if (inventory.getInventory().size() == 0) {
+            return null;
```
Comment (bw55555) [inline src/main/java/save/SaveInventory.java:16]: Try not to return null, return empty string instead

File: src/main/java/save/SaveEntityId.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 package save;
 
 public enum SaveEntityId {
-
+    INVENTORY_ID
```
Comment (bw55555) [inline src/main/java/save/SaveEntityId.java:4]: perhaps name as INVENTORY instead of INVENTORY_ID?

**Review by bw55555 — CHANGES_REQUESTED** [review]

**Review by bw55555 — APPROVED** [review]
Looks good! Methods tested and looks clean!

**Review by mahirhassan-uoft — APPROVED** [review]

**Review by Yasutakahi — APPROVED** [review]
code looks greate and changes make sense!

**Review by mahirhassan-uoft — APPROVED** [review]

**Review by Yasutakahi — DISMISSED** [review]
looks great!!

**Review by passionfruitjello — DISMISSED** [review]

File: src/main/java/objects/inventory/Inventory.java [hunk 3]
```diff
@@ -97,11 +100,67 @@ public boolean removeItem(int i){
...
+            this.addItem(i);
+        }
+    }
+
+    /**
+     * @param itemType type of item
+     * @param level level of item
+     * @return Item object
+     */
+    public static Item createItem(String level, String itemType) {
```
Comment (bw55555) [inline src/main/java/objects/inventory/Inventory.java:146]: perhaps should not be static

**Review by bw55555 — APPROVED** [review]
refactor changes in a later pull request

**Review by marjaleenakiik — APPROVED** [review]
Changes make sense to fix the warnings and bugs.

**Pull request merged**